pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -270,26 +278,26 @@ func getFinishAndExpireTime(j *batch.Job) (*time.Time, *time.Time, error) {
 	if !needsCleanup(j) {
 		return nil, nil, fmt.Errorf("job %s/%s should not be cleaned up", j.Namespace, j.Name)
 	}
-	finishAt, err := jobFinishTime(j)
+	t, err := jobFinishTime(j)
 	if err != nil {
 		return nil, nil, err
 	}
-	finishAtUTC := finishAt.UTC()
-	expireAtUTC := finishAtUTC.Add(time.Duration(*j.Spec.TTLSecondsAfterFinished) * time.Second)
-	return &finishAtUTC, &expireAtUTC, nil
+	finishAt := t.Time
+	expireAt := finishAt.Add(time.Duration(*j.Spec.TTLSecondsAfterFinished) * time.Second)
+	return &finishAt, &expireAt, nil
 }
 
-func timeLeft(j *batch.Job, since *time.Time) (*time.Duration, error) {
+func timeLeft(j *batch.Job, since *time.Time) (*time.Duration, *time.Time, error) {
 	finishAt, expireAt, err := getFinishAndExpireTime(j)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	if finishAt.UTC().After(since.UTC()) {
+	if finishAt.After(*since) {
 		klog.Warningf("Warning: Found Job %s/%s finished in the future. This is likely due to time skew in the cluster. Job cleanup will be deferred.", j.Namespace, j.Name)
 	}
-	remaining := expireAt.UTC().Sub(since.UTC())
+	remaining := expireAt.Sub(*since)
```

Comment:
Using UTC is intentional so we avoid problems like daylight saving.

Reply:
`.UTC()` only sets the location used when converting the time to a string, it doesn't impact the calculations that happen on the value which is always stored as the time elapsed since Jan 1 1970 UTC, so this is a noop for what we want to do.